php/lang/correctness/empty-with-boolean-expression.yaml

```diff
@@ -0,0 +1,16 @@
+rules:
+- id: empty-with-boolean-expression
+  pattern-either:
+  - pattern: |
+      empty($A && $B)
+  - pattern: |
+      empty($A || $B)
+  message: >-
+    Calling `empty` on a boolean expression may be an indication that a parenthesis is misplaced.
+  metadata:
+    category: correctness
+    technology:
+    - php
+    confidence: LOW
+  languages: [php]
+  severity: WARNING
```

Comment:
Maybe worth lowering severity to `INFO`?

Reply:
I didn't give much thought to what the severity should be, but this rule does seem to reliably find actual bugs, in my experience, so I think `WARNING` is accurate.